lib/stripe/error_object.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Stripe
+  class ErrorObject < StripeObject
+    # Unlike other objects, we explicitly declare getter methods here. This
+    # is because the API doesn't return `null` values for fields on this
+    # object, rather the fields are omitted entirely. Not declaring the getter
+    # methods would cause users to run into `NoMethodError` exceptions and
+    # prevent any sort of generic error handling.
+
+    def charge
+      @values[:charge]
+    end
+
+    def code
+      @values[:code]
+    end
+
+    def doc_url
+      @values[:doc_url]
+    end
+
+    def message
+      @values[:message]
+    end
+
+    def param
+      @values[:param]
+    end
+
+    def payment_intent
+      @values[:payment_intent]
+    end
+
+    def payment_method
+      @values[:payment_method]
+    end
+
+    def setup_intent
+      @values[:setup_intent]
+    end
+
+    def source
+      @values[:source]
+    end
+
+    def type
+      @values[:type]
+    end
+
+    # For OAuth errors
+
+    def error
```

Comment:
In stripe-go we ended up calling these `OAuthError` and `OAuthErrorDescription` (i.e. prefixed with "oauth"): https://github.com/stripe/stripe-go/pull/879

Might be a good idea here too given that the names are somewhat generic and if you didn't see that they were in their own section, you could easily make the mistake that they're supposed to be used outside of OAuth-related context.

Reply:
👍  Good idea.